spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdReporter.java

```diff
@@ -112,66 +88,71 @@ private void reportCounters(SortedMap<String, Counter> counters, DatagramSocket
-                    metricFormatter.buildMetricString(name, "max", snapshot.getMax(), TIMER),
-                    metricFormatter.buildMetricString(name, "mean", snapshot.getMean(), TIMER),
-                    metricFormatter.buildMetricString(name, "min", snapshot.getMin(), TIMER),
-                    metricFormatter.buildMetricString(name, "stddev", snapshot.getStdDev(), TIMER),
-                    metricFormatter.buildMetricString(name, "p50", snapshot.getMedian(), TIMER),
-                    metricFormatter.buildMetricString(name, "p75", snapshot.get75thPercentile(), TIMER),
-                    metricFormatter.buildMetricString(name, "p95", snapshot.get95thPercentile(), TIMER),
-                    metricFormatter.buildMetricString(name, "p98", snapshot.get98thPercentile(), TIMER),
-                    metricFormatter.buildMetricString(name, "p99", snapshot.get99thPercentile(), TIMER),
-                    metricFormatter.buildMetricString(name, "p999", snapshot.get999thPercentile(), TIMER)
-            );
+            List<String> metrics = new ArrayList<>();
+            addIfMatches(metrics, MetricAttribute.COUNT, metricFormatter.buildMetricString(name, MetricAttribute.COUNT.toString(), histogram.getCount(), GAUGE));
+            addIfMatches(metrics, MetricAttribute.MAX, metricFormatter.buildMetricString(name, MetricAttribute.MAX.toString(), snapshot.getMax(), TIMER));
+            addIfMatches(metrics, MetricAttribute.MEAN, metricFormatter.buildMetricString(name, MetricAttribute.MEAN.toString(), snapshot.getMean(), TIMER));
+            addIfMatches(metrics, MetricAttribute.MIN, metricFormatter.buildMetricString(name, MetricAttribute.MIN.toString(), snapshot.getMin(), TIMER));
+            addIfMatches(metrics, MetricAttribute.STDDEV, metricFormatter.buildMetricString(name, MetricAttribute.STDDEV.toString(), snapshot.getStdDev(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P50, metricFormatter.buildMetricString(name, MetricAttribute.P50.toString(), snapshot.getMedian(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P75, metricFormatter.buildMetricString(name, MetricAttribute.P75.toString(), snapshot.get75thPercentile(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P95, metricFormatter.buildMetricString(name, MetricAttribute.P95.toString(), snapshot.get95thPercentile(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P98, metricFormatter.buildMetricString(name, MetricAttribute.P98.toString(), snapshot.get98thPercentile(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P99, metricFormatter.buildMetricString(name, MetricAttribute.P99.toString(), snapshot.get99thPercentile(), TIMER));
+            addIfMatches(metrics, MetricAttribute.P999, metricFormatter.buildMetricString(name, MetricAttribute.P999.toString(), snapshot.get999thPercentile(), TIMER));
+            send(socket, metrics.toArray(new String[metrics.size()]));
```

Comment:
`metrics.toArray(new String[0])` is recommended for newer Java versions and more performant starting with Java 8.

Reply:
Changed it accordingly.